docs/distribution/local-repository.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: developer-doc
+title: Local Repository
+category: distribution
+tags: [distribution, project-manager, offline, local]
+order: 7
+---
+
+# Local Repository
+
+A `LocalReleaseProvider` is implemented that allows to install components from
+local (offline) repositories. This functionality can be used to allow installing
+bundled components.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Local Repository Structure](#local-repository-structure)
+  - [General Repository Structure](#general-repository-structure)
+  - [Engine Repository Structure](#engine-repository-structure)
+  - [GraalVM Repository Structure](#graalvm-repository-structure)
+- [Usage in Project Manager](#usage-in-project-manager)
+
+<!-- /MarkdownTOC -->
+
+## Local Repository Structure
+
+This section explains how a local repository has to be structured in order to
+work with `LocalReleaseProvider`.
+
+### General Repository Structure
+
+In general the local repository should be a separate directory, not containing
+any other files. For each provided release it should contain a directory called
+after the release's tag. That inner directory should just contain assets
+associated with that release.
```

Comment:
Reading this my first thought was, that the local repository should look like
```
└── my_local_repository/
    └── 0.1.2/
        ├── enso-0.1.2-rc.9/
        │   ├── enso-engine-0.1.2-rc.9-linux-amd64.tar.gz
        │   └── manifest.yaml
        └── vm-20.2.0/
            └── graalvm-ce-java11-linux-amd64-20.2.0.tar.gz
```
But when I look at the further examples I'm not so sure. Perhaps we should define what exactly is "repository' (is it repository of many different components, or only of one?), and what are release's tags (I see no tags nor releases currently in enso repository).

Reply:
Ok, now I see that the repository contains just one "component" (engine or graalvm). I think if you add a note at the beginning, it should be ok.

Reply:
The tags are actually release tags, our current repo does not have any currently (as we do not have an official release yet), but you can see them on https://github.com/enso-org/enso-staging/releases

Similarly, for GraalVM the tags correspond to tags here https://github.com/graalvm/graalvm-ce-builds/releases

I will add a note saying that a single repository should contain just one component and a more detailed example for the PM in a moment.